### ISSUE TITLE
Drafts keep reappearing in topic groups on other devices

### ISSUE THREAD

**Issue by Inurosen** [telegramdesktop/tdesktop#26236]
### Steps to reproduce

1. Write a message in a topic group.
2. Wait for it to become draft/
3. Send the message.
4. After some time open Telegram on other device.
5. There will be the latest draft in topic's chat box. The sent message is visible in conversation.
7. Delete the draft.
8. After some time open Telegram on third device.
9. There will be the same draft. The sent message is visible in conversation

### Expected behaviour

Drafts of sent messages in topics don't reappear.

### Actual behaviour

Drafts of sent messages in topics reappear on all devices connected to senders account.

### Operating system

Any

### Version of Telegram Desktop

4.8.1

### Installation source

Static binary from official website

**Comment by bastiaand** [comment]
> Noticed server recently began aggressively pushing old drafts onto client's throat

That may be true, but if you have a draft/unsent message in TG web it will appear every ? seconds in your TG desktop session.  Just get rid of the draft/unsent message regardless of where it is and the problem should go away.  

Fun stuff.

**Comment by sempamavin** [comment]
I have the same issue in the telegram on windows and android devices.
I'm sending a message, but Telegram prompts me to edit the sent message as a draft.

**Comment by taifu** [comment]
Same here. Hateful problem.

**Comment by gabiiel** [comment]
https://github.com/telegramdesktop/tdesktop/issues/2229
At least 7 year old problem, I deleted the message on all active devices but it still was updating with this annoyance every 10 sec
Send the message, it gets back, try to leave another, smaller one, it gets back and deletes legitimate stuff, not even "delete all cloud draft" worked until I removed all the devices

Just give me the option to not use cloud draft, if I want to send a message across platforms, I can use the "saved messages tab" or simply rewrite the stuff, if you can't do it right, give me the option  to not have your broken feature

**Comment by RobT2012** [comment]
Here's how I fixed it. I was using TG on my Android phone, and TG web in Chrome browser on a Linux machine. What appears to have been the problem was having multiple open pages with TG Web on the browser.  After I closed all the open TG Web instances, except for only one, then I could delete the draft message and it finally went away for good. I left the Android TG app open when I was closing TG Web sessions, so it appears to not matter if the app is still connected while using the web app, it's only a problem if there are multiple TG web pages open. The problem may happened with another type of browser such as FireFox but I've not tested it to find out.

**Comment by vilmaera** [comment]
Yeah. Even faster way to stop is this: 

I have 2 devices: chrome and mobile.

1. Go to your mobile, FORCE STOP your Telegram.
2. Go to your PC and delete the message
3. Paste something quick and delete it
4. It helped me

**Comment by RobT2012** [comment]
I can confirm that the recurring "draft" message is reproducible by opening more than one instance of TG connected to the same account. Some posts will end up popping up as draft posts long after they were committed or deleted, there's no way to remove the draft messages except by closing all instances of TG except for one, and deleting the draft on the remaining open TG instance. My best guess is that which posts end up becoming recurring drafts depends on how long they were left open before committing them or deleting them. The ones of short duration (less than a minute maybe) probably are fine, but the ones left unfinished for a few minutes seem to become recurring drafts.

**Comment by lehonmikko** [comment]
Same thing for me. Draft won delete.. I will try some of your suggestions.

**Comment by thuongnvblockodyssey** [comment]
drafts is a very annoying feature on telegram. I often open telegram on multiple devices at the same time. And drafts automatically syncs between devices.

**Comment by Aokromes** [comment]
drafts aren't annoying, they are usefull, bugged drafts are annoying.

**Comment by jsphwllng** [comment]
An option to disable cloud drafts on your account would be useful here.

**Comment by foxmajik** [comment]
I confirm that this problem exists across the web, desktop (windows) and Android versions.

**Comment by JoseHernandez88** [comment]
yeah I have this issue as well

**Comment by jstafford5380** [comment]
Yep definitely have this problem and it's not going away no matter how many times I delete it. It has completely taken over a group chat. I have telegram on probably a dozen devices that I'm now going to have to disconnect and reconnect and pray to god it goes away

**Comment by hurtonypeter** [comment]
Any update on this?

**Comment by eric2788** [comment]
same here. It's so annoying

**Comment by nvthuong1996** [comment]
Drafts are an extremely annoying feature in telegram. Please correct it

**Comment by jstafford5380** [comment]
I've had the same text on every device that I have to erase before sending
a new message -- for the last 6 months.

**Comment by RememberTheAir** [comment]
Not sure if this is the same issue, but drafts keep coming back to my input box over and over and over, in non-forum chats and in single topics. Clearing them from desktop doesn't work - the next time you open it the draft will be back. Clearing them from another client won't work, they will come back too. I've ignored this for a few months, but now I have way too many topics with ghost drafts to ignore the issue :/

**Comment by tugruul** [comment]
Same shit happens  to me. İ got rid off that bug  donno how. But  it came  to my  browser again. İ  deleted all cloud drafts from settings, but  nıthing  happened.  After that  terminated all devices still  nothing.  Seriosly  it  is very annoying that  all the time  before  group message  u have to delete draft  after that  u  can send message.

**Comment by krischik** [comment]
> Get rid of your draft/unsent messages on other devices and your issue may go away.

That's not feasible as some of those devices are physically several kilometre apart so it's impossible to delete them all in time before the draft reappears on all devices.

**Comment by krischik** [comment]
> I have 2 devices: chrome and mobile.

Doesn't work if you have more then two devices.

**Comment by krischik** [comment]
> drafts aren't annoying, they are useful, bugged drafts are annoying.

I rather have no drafts then bugged drafts.

**Comment by danslo** [comment]
Agreed, the only way to fix this is by terminating the other session.
But there's a limit to how many times you can add devices, so it's a hacky workaround that doesn't work half of the time.

All I ask for is a toggle to disable drafts.

**Comment by krischik** [comment]
The reason why I'm here: Drafts have become even more aggressive.

They now delete what you have typed and replace it with a version of the same message from a few minutes ago. Now I have to prepare my message in a text editor and copy paste it and quickly press send.

Of course within seconds the older version of the draft appears again.

Telegram: If you can't get it to work properly remove drafts.

**Comment by Aokromes** [comment]
myself i don't have seen any unwanted draft for long time

**Comment by tws-rdelatorre** [comment]
> myself i don't have seen any unwanted draft for long time

I have, quite a few times over the past year or so, last time this very morning.

If drafts was a feature that could be turned off, I'd turn it off for sure and personally wouldn't miss it.

**Comment by hezamu** [comment]
This issue keeps appearing constantly.

Cloud sync for drafts is a misfeature. Please remove it, or at least give me an option to disable it, preferably opt-in.

**Comment by baltika10** [comment]
I've found to "partially" solve the problem when you don't have access to the PC with the opened web telegram, is to update the draft value to something with one symbol (for example dot) from the phone, so it's easier to remove it later, when you need to write something.
And I agree, this is useless annoying feature

**Comment by krischik** [comment]
> We therefore assume that the user has lost interest

No we have not lost interest. We have lost hope. Hope that Telegram will ever fix this issue. I suspect that Telegram introduced a feature which exceeds the skill of their developer and are now to proud to admit defeat and remove the feature again.

**Comment by danslo** [comment]
>We therefore assume that the user has lost interest

We switched to Signal, which doesn't have this problem. 🤷

**Comment by kondr1** [comment]
I see that just now

**Comment by 916253** [comment]
This problem still affects me

**Comment by pacovila** [comment]
This problem still affects me as of 2026

**Comment by fmillion-mnsu** [comment]
I'm surprised this continues to be an issue years later. The fix is simple: either completely get rid of cloud drafts (they're honestly not *that* useful) or, as many have suggested, make it a toggle. Whatever logic is checking for or receiving cloud drafts--that functionality should go away when the toggle is off. 

The *real* solution would just be to remove any cloud-saved draft whenever any client sends a message as that user. I can't think of any use case for "I sent this message but I still want it syncing as a draft". The functionality is broken by design - if I start a message on my desktop, then open my phone and see the draft, I can even *send* the draft and the draft keeps re-appearing. Apparently they didn't design in any way for a client to be told to *delete* its draft; only to *send* its current draft. So what I gather happens is client A has a draft, client B logs in and gets that draft, client B sends the draft as a message, but client A still broadcasts the fact that it has a draft. So client B gets the draft again...and again...

If this can't be fixed, then just get rid of it. It doesn't even work as intended right now.

**Comment by vozgocom** [comment]
This issue/bug still continues. It has been years and still no updates.

Problem cant be solved without exiting from all connections. This makes time-loss.

**Comment by ropinheiro** [comment]
Also happening with me, and I can't afford being constantly closing sessions. I work in multiple computers for multiple contexts, plus mobilephone, and sometimes I am just writing something in TG, get interrupted, go back to what I was writing in TG only to find it as a draft eternally synched even days after I wrote it. It was not even intended to be a draft, TG just decided it to be once I am some seconds away from the window...
